Added workflow for creating and pushing git tag (update actions/checkout to the version 4)

Nikita-Smirnov-Exactpro opened the pull request.

OptimumCode commented on .github/workflows/compaund-git-tag-push.yml:
```diff
@@ -0,0 +1,32 @@
+          echo "description ${{ inputs.tagDescription }}"
+      - name: Create and push tag
+        run: |
+          git config --global user.name 'Github Actions'
+          git config --global user.email 'exactpro@exactpro.com'
```
This is usually the email that is used for auto-commits in GitHub:
name = 'github-actions'
email = '41898282+github-actions[bot]@users.noreply.github.com'

Probably, we should use this one as well

OptimumCode reviewed (approved):
Just a small note: the PR is related to adding a new create-tag step. But we also updated the dependencies. It is not a good practice to mix changes...

Nikita-Smirnov-Exactpro commented:
> Just a small note: the PR is related to adding a new create-tag step. But we also updated the dependencies. It is not a good practice to mix changes...

I noticed the warning related to https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/
and added updates into this change.
Mix changes is a bad practice because revert operation will be more difficult, also, we apply not declared change?

OptimumCode commented:
> Mix changes is a bad practice because revert operation will be more difficult, also, we apply not declared change?

If this is the question about why mixing changes is a bad practice then yes and yes - the original purpose of the PR is to add workflow for adding tags. And it should contain only changes that correspond to that. If we need to update dependencies or etc we can open another PR...
Mixing changes in a single PR also does not help automated workflows, e.g. automatically collecting release notes

Nikita-Smirnov-Exactpro commented:
> Mixing changes in a single PR also does not help automated workflows, e.g. automatically collecting release notes
It is a good point, thank you )
